syntax = "proto3";

package eddid.quote;
option java_multiple_files = true;

enum QuoteMsgType
{
    UNKNOWN_MSG               = 0;
    
    HEARTBEAT_REQ             = 1;              // 心跳同步请求
    HEARTBEAT_RSP             = 2;              // 心跳同步应答
                                                                
    VELOCITY_REQ              = 3;              // 测速请求
    VELOCITY_RSP              = 4;              // 测速应答
    
    LOGIN_REQ                 = 5;              // 用户登录请求
    LOGIN_RSP                 = 6;              // 用户登录应答
    LOGOUT_REQ                = 7;              // 用户登出请求
    LOGOUT_RSP                = 8;              // 用户登出应答
    
    SYNC_INSTR_REQ            = 9;              // 合约同步请求
    SYNC_INSTR_RSP            = 10;             // 合约同步应答
    
    QUERY_KLINE_REQ           = 11;             // K线查询请求
    QUERY_KLINE_RSP           = 12;             // K线查询应答 
    QUERY_TICK_REQ            = 13;             // TICK查询请求
    QUERY_TICK_RSP            = 14;             // TICK查询应答 
    QUERY_TRADE_TICK_REQ      = 15;             // 逐笔成交查询请求
    QUERY_TRADE_TICK_RSP      = 16;             // 逐笔成交查询应答
    QUERY_KLINE_MIN_REQ       = 17;             // 查询分时数据 
    QUERY_KLINE_MIN_RSP       = 18;             // 查询分时数据应答 
    QUERY_FIVEDAYS_KLINEMIN_REQ = 19;           // 五日分时数据查询请求
    QUERY_FIVEDAYS_KLINEMIN_RSP = 20;           // 五日分时数据查询应答
    
    QUERY_TRADE_DAY_REQ       = 25;             // 交易日历查询请求
    QUERY_TRADE_DAY_RSP       = 26;             // 交易日历查询应答 
    
    QUERY_EXCHANGE_RATE_REQ   = 27;             // 查询汇率信息
    QUERY_EXCHANGE_RATE_RSP   = 28;             // 查询汇率应答

    QUERY_INSTR_SORT_REQ      = 29;             // 查询合约排序
    QUERY_INSTR_SORT_RSP      = 30;             // 查询合约排序
    
    SUBSCRIBE_REQ             = 37;             // 行情订阅请求
    SUBSCRIBE_RSP             = 38;             // 行情订阅应答
    UNSUBSCRIBE_REQ           = 39;             // 取消订阅请求
    UNSUBSCRIBE_RSP           = 40;             // 取消订阅应答
    SUBSCRIBE_KLINE_REQ       = 41;             // 订阅K线请求
    SUBSCRIBE_KLINE_RSP       = 42;             // 订阅K线请求应答
    UNSUBSCRIBE_KLINE_REQ     = 43;             // 取消订阅K线请求
    UNSUBSCRIBE_KLINE_RSP     = 44;             // 取消订阅K线请求应答
    UNSUBSCRIBE_KLINE_MIN_REQ = 45;             // 取消订阅分时数据请求
    UNSUBSCRIBE_KLINE_MIN_RSP = 46;             // 取消订阅分时数据应答
    SUBSCRIBE_KLINE_MIN_REQ   = 47;             // 订阅分时请求
    SUBSCRIBE_KLINE_MIN_RSP   = 48;             // 订阅分时请求应答
    SUBSCRIBE_TRADE_TICK_REQ   = 49;            // 订阅逐笔请求
    SUBSCRIBE_TRADE_TICK_RSP   = 50;            // 订阅逐笔请求应答
    UNSUBSCRIBE_TRADE_TICK_REQ = 51;            // 取消订阅逐笔数据请求
    UNSUBSCRIBE_TRADE_TICK_RSP = 52;            // 取消订阅逐笔数据应答
   
    PUSH_OPEN                 = 53;             // 开盘通知
    PUSH_CLOSE                = 54;             // 收盘通知 
    PUSH_BASIC                = 55;             // 推送行情静态数据 
    PUSH_SNAPSHOT             = 56;             // 推送行情快照数据
    PUSH_ORDER_BOOK           = 57;             // 推送行情盘口数据
    PUSH_TRADE_DATA           = 58;             // 推送行情逐笔成交数据                  
    PUSH_INSTR_INFO           = 59;             // 推送行情合约信息数据
    PUSH_KLINE                = 60;             // 推送行情K线数据
    PUSH_KLINE_MIN            = 61;             // 推送行情分时数据
    PUSH_TRADE_DAY            = 62;             // 推送交易日历
    PUSH_TICK_DATA            = 63;             // 推送tick数据
    
    SYNC_BASIC_REQ            = 64;              // 静态数据同步请求
    SYNC_BASIC_RSP            = 65;              // 静态数据同步应答
    
    SNAPSHOT_REQ              = 66;              // 快照数据请求
    SNAPSHOT_RSP              = 67;              // 快照数据应答
    
    LOAD_HIS_KLINE_REQ        = 68;              // 历史K线查询请求（计算服务加载历史K线数据）
    LOAD_HIS_KLINE_RSP        = 69;              // 历史K线查询应答（计算服务加载历史K线数据）

    QUERY_TRADE_STATUS_REQ    = 70;              // 查询品种交易状态查询请求
    QUERY_TRADE_STATUS_RSP    = 71;              // 查询品种交易状态查询应答    
    PUSH_TRADE_STATUS         = 72;              // 推送品种交易状态查询   
   
    // 手机专用命令字 
    START_CHART_DATA_REQ      = 500;            // 请求图表数据
    START_CHART_DATA_RSP      = 501;            // 请求图表数据应答 
    STOP_CHART_DATA_REQ       = 502;            // 图表数据停止请求
    STOP_CHART_DATA_RSP       = 503;            // 图表数据停止应答
}

// 合约同步方式
enum SyncInstrMsgType
{
    UNKNOWN_SYNC_INSTR      = 0;
    ALL_INSTR               = 1;       // 全量方式同步
    INCREMENT_INSTR         = 2;       // 增量方式同步
}

// K线周期定义
enum KLinePeriodType
{
    UNKNOWN_PERIOD   = 0;
    ONE_SECONDS      = 1;       // 1s
    THREE_SECONDS    = 2;       // 3s
    FIVE_SECONDS     = 3;       // 5s
    TEN_SECONDS      = 4;       // 10s
    FIFTEEN_SECONDS  = 5;       // 15s
    TWENTY_SECONDS   = 6;       // 20s
    THIRTY_SECONDS   = 7;       // 30s
    MINUTE           = 8;       // 1min
    TWO_MIN          = 9;       // 2min
    THREE_MIN        = 10;      // 3min
    FIVE_MIN         = 11;      // 5min
    TEN_MIN          = 12;      // 10min.
    FIFTEEN_MIN      = 13;      // 15min
    THIRTY_MIN       = 14;      // 30min
    HOUR             = 15;      // 1hour
    TWO_HOUR         = 16;      // 2hour
    FOUR_HOUR        = 17;      // 4hour
    DAY              = 18;      // 1day
    WEEK             = 19;      // 1week
    MONTH            = 20;      // 1month
    SEASON           = 21;      // 1season
    YEAR             = 22;      // 1year
    MIN_BY_MIN       = 23;      // 分时（图）
    MAX_PERIOD_TYPE  = 24;
}

// 获取K线的方式 
enum QueryKLineMsgType
{
    UNKNOWN_QUERY_KLINE = 0;
    BY_DATE_TIME        = 1;          // 按日期时间获取 
    BY_VOL              = 2;          // 按量获取(此方式下，会从最新时间开始向后数取相就数量的bar)
}

// 获取K线的方向
enum QueryKLineDirectType
{
    UNKNOWN_QUERY_DIRECT = 0;
    WITH_FRONT           = 1;              // 指定时间向前查询
    WITH_BACK            = 2;              // 指定时间向后查询
}

// 订阅消息类型
enum SubscribeMsgType
{
    UNKNOWN_SUB       = 0;
    SUB_WITH_INSTR    = 1;                     // 按合约订阅
    SUB_WITH_PRODUCT  = 2;                     // 按品种订阅
    SUB_WITH_MARKET   = 3;                     // 按市场订阅
    SUB_WITH_MSG_DATA = 4;                     // 按消息类型（订阅静态数据， 订阅快照，订阅盘口，订阅逐笔……）
}

// 订阅子消息类型
enum SubChildMsgType
{
    UNKNOWN_SUB_CHILD = 0;
    SUB_BASIC        = 1;                    // 订阅静态数据 
    SUB_SNAPSHOT     = 2;                    // 订阅快照
    SUB_ORDER_BOOK   = 3;                    // 订阅盘口
    SUB_TRADE_DATA   = 4;                    // 订阅逐笔
}

// 港股成交类型
enum StockTradeType
{
    UNKNOWN_STOCK_TRADE = 0;            // 自动搓和（对盘）
    P                = 1;                   // 开市前成交盘
    M                = 2;                   // 非自动搓和（对盘）
    Y                = 3;                   // 同一证券商自动搓和（对盘）
    X                = 4;                   // 同一证券商非自动搓和（对盘）
    D                = 5;                   // 碎股交易
    U                = 6;                   // 竞价交易
    Z                = 7;                   // 交易的一方不是香港交易所的成员，属于场外交易
}

// 期货成交类型
enum FutureTradeType
{
    UNKNOWN_FUTURE_TRADE   = 0;              // None
    FUTURE_TRADE_TYPE_P    = 1;                   // Printable
    FUTURE_TRADE_TYPE_O    = 2;                   // Occured at cross
    FUTURE_TRADE_TYPE_R    = 3;                   // reported trade
}

// 请求交易日历方式
enum QueryTradeDayType
{
    UNKNOWN_QUERY_TRADE_DAY = 0;
    RANGE_QUERY = 1;
    PLUS_QUERY = 2;
}

enum QueryInstrSortType
{
    UNKNOWN_QUERY_INSTR_SORT_TYPE = 0;
    ASCENDING_ORDER  = 1;   // 升序
    DESCENDING_ORDER  = 2;  // 降序
}

enum QueryInstrSortCondition
{
    UNKNOWN_QUERY_INSTR_SORT_CONDITION = 0;
    BY_CODE = 1;                 // 合约代码
    BY_PRE_CLOSE = 2;            // 昨收价
    BY_PRE_SETTLEMENT_PRICE = 3; // 上次结算价
    BY_PRE_OPEN_INTERREST = 4;   // 昨持仓量
    BY_OPEN = 5;                 // 开盘价
    BY_HIGH = 6;                 // 最高价
    BY_LOW = 7;                  // 最低价
    BY_LAST = 8;                 // 最新价
    BY_VOLUME = 9;               // 成交量
    BY_RISE_FALL = 10;           // 涨跌
    BY_R_F_RATIO = 11;           // 涨跌幅
    BY_OPEN_INTERREST = 12;      // 持仓量
    BY_ASK_VOL = 13;             // 卖盘量
    BY_BID_VOL = 14;             // 买盘量
    BY_FIRST_ASK_PRICE = 15;     // 当前盘口价 买1价格
    BY_FIRST_BID_PRICE = 16;     // 当前盘口价 卖1价格
}
